src/Umbraco.Core/Models/ISimpleContentType.cs

```diff
@@ -46,6 +46,11 @@ public interface ISimpleContentType
     /// </summary>
     bool IsElement { get; }
 
+    /// <summary>
+    ///     Gets the aliases of the content types that compose this content type.
+    /// </summary>
+    IEnumerable<string> CompositionAliases => Enumerable.Empty<string>();
```

Comment:
I've added a default implementation to the interface in order for this to not be a breaking change.
Not sure if this is something we want, or if it should just be removed and the change introduced in a major.

Reply:
It still gonna be breaking change. 
https://learn.microsoft.com/en-us/dotnet/standard/library-guidance/breaking-changes

> Adding anything to an interface will cause existing types that implement it to fail. An abstract base class allows you to add a default virtual implementation.

Reply:
That used to be the case until C# 8 introduced default implementations in interfaces.
This change should be both source and binary compatible, and this was the case in the tests I did.

https://devblogs.microsoft.com/dotnet/default-implementations-in-interfaces/
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/default-interface-methods

> Default interface methods enable an API author to add methods to an interface in future versions without breaking source or binary compatibility with existing implementations of that interface.


---
The reason why you might not want to have a default implementation is that of course it doesn't enforce that the consuming types properly implement the new behavior.

Reply:
That is interesting that Microsoft has 2 pages in docs saying 2 different thing! As first which I linked state this change would be breaking but based on other pages which you linked seems like it is not. So I am resolving my comment as I learned new thing about Microsoft docs being not clear 😂